src/scripts/main.js

```diff
@@ -1,3 +1,324 @@
+  for (const row of table.rows) {
+    if (row.classList.contains('active')) {
+      row.classList.remove('active');
+    }
+  }
+};
+
+tBody.addEventListener('click', (e) => {
+  const target = e.target.closest('tr');
+
+  if (target) {
+    deselectRows();
+    target.classList.add('active');
+  }
+});
+// #endregion
+
+// #region FORM_NEW_EMPLOYEE
+// DATA FOR TJE FORM
+const offices = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+// CREATING FORM ELEMENTS
+const formatLabelText = (inputName) => {
+  return `${inputName.charAt(0).toUpperCase()}${inputName.substring(1)}: `;
+};
+
+const createInput = (inputName, inputType) => {
+  const input = Object.assign(document.createElement('input'), {
+    name: inputName,
+    type: inputType,
+  });
+  const label = document.createElement('label');
+
+  input.setAttribute('data-qa', inputName);
+
+  label.textContent = formatLabelText(inputName);
+  label.append(input);
+
+  return label;
+};
+
+const createSelect = (inputName, options) => {
+  const select = Object.assign(document.createElement('select'), {
+    name: inputName,
+  });
+
+  select.setAttribute('data-qa', inputName);
+
+  Array.from(options)
+    .sort()
+    .forEach((option) => select.append(new Option(option, option)));
+
+  const label = document.createElement('label');
+
+  label.textContent = formatLabelText(inputName);
+  label.append(select);
+
+  return label;
+};
+
+const createButton = (btnText, type) => {
+  const button = document.createElement('button');
+
+  button.setAttribute('type', type);
+  button.textContent = btnText;
+
+  return button;
+};
+
+const formElements = [
+  createInput('name', 'text'),
+  createInput('position', 'text'),
+  createSelect('office', offices),
+  createInput('age', 'number'),
+  createInput('salary', 'number'),
+  createButton('Save to table', 'submit'),
+];
+
+// CREATE FORM AND ADD ELEMENTS
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+formElements.forEach((element) => form.append(element));
+document.addEventListener('DOMContentLoaded', () => document.body.append(form));
+
+// FORM VALIDATION
+const positionValidatorFunction = (value) =>
+  !value
+    ? { validated: false, message: `Position is a required field!` }
+    : { validated: true };
+
+const nameValidatorFunction = (value) =>
+  !value || value.length < 4
+    ? { validated: false, message: 'Name should be more than 4 chars long' }
+    : { validated: true };
+
+const ageValidatorFunction = (value) =>
+  value < 18 || value > 90
+    ? { validated: false, message: 'Age must be between 18 and 90' }
+    : { validated: true };
+
+const getValidatorFunction = (value, key) => {
+  const validators = {
+    position: positionValidatorFunction,
+    name: nameValidatorFunction,
+    age: ageValidatorFunction,
+  };
+
+  return validators[key]?.(value) || { validated: true };
+};
+
+const validateForm = (data) => {
```

Comment:
The `validateForm` function uses `Array.from` to convert the FormData object to an array. This is unnecessary because FormData objects are already iterable. You can simplify this code by directly using a `for...of` loop or the `forEach` method on the FormData object.

Reply:
@mateacademy-ai-mentor your comment is out of context. You make a good point, if you ignore next three lines of code, which you should not do.